DUnitX.WeakReference.pas

```diff
@@ -62,23 +66,26 @@ interface
   ///  it. It implements IInterface so the object can also be used just like
   ///  any normal reference counted objects in Delphi.
   TWeakReferencedObject = class(TObject, IInterface, IWeakReferenceableObject)
+  private const
+    objDestroyingFlag = Integer($80000000);
   protected
+  {$IFNDEF AUTOREFCOUNT}
     FRefCount: Integer;
+  {$ENDIF}
     FWeakReferences : TList<Pointer>;
-    FDestroying : boolean;
-    FDestroyed : boolean;
     function QueryInterface(const IID: TGUID; out Obj): HResult; stdcall;
     function _AddRef: Integer; stdcall;
     function _Release: Integer; stdcall;
     procedure AddWeakRef(value : Pointer);
     procedure RemoveWeakRef(value : Pointer);
-    function GetRefCount : integer;
+    function GetRefCount : integer; inline;
```

Comment:
Shouldn't this method be ifdef's as well? It uses FRefCount which is ifdef out?

Reply:
It uses FRefCount from TObject under NEXTGEN.

Reply:
Ah right, I didn't realise it was protected, I thought it was private.